flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/operators/AbstractAsyncStateStreamOperatorV2.java

```diff
@@ -260,6 +262,15 @@ public void processWatermark(Watermark mark) throws Exception {
         asyncExecutionController.processNonRecord(() -> super.processWatermark(mark));
     }
 
+    protected void emitWatermarkDirectly(Watermark mark) throws Exception {
```

Comment:
NIT: now that the watermark can be emitted in a future in this method it does not seem directly. I wonder if we could rename to `emitWatermark`.

Reply:
As Zakelly's suggestion, I removed `emitWatermark` in `AbstractAsyncStateStreamOperatorV2`.